Integrations/integration-AzureAD.yml

```diff
@@ -0,0 +1,157 @@
+        var res = http(
+            url,
+            {
+                Method:  method,
+                Headers: {
+                    'Content-Type': ['application/x-www-form-urlencoded'],
+                    'Authorization' : token? [token_type + ' ' + token] : undefined
+                }, //TODO add headers
+                Body : body? encodeToURLQuery(body).substr(1) : undefined
+            },
+            params.insecure,
+            params.proxy
+        );
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+        return res.Body;
+    };
+
+    var client_id = params.client_id;
+    var client_secret = params.secret;
+    var login_url = params.login_url.replace(/[\/]+$/, '');
+    var tenant_domain = params.domain.replace(/[\/]+$/, '');
+
+    var token;
+    var token_type;
+
+    var getToken = function(){
+        request_url = login_url + '/'+tenant_domain + '/oauth2/v2.0/token';
+        bodyvals = {
+            'client_id': client_id,
+            'client_secret': client_secret,
+            'grant_type': 'client_credentials',
+            'scope': 'https://graph.microsoft.com/.default'
+        };
+        res = sendRequest(request_url, 'POST', {}, bodyvals);
+        try {
+            result = JSON.parse(res);
+        } catch (err) {
+            throw 'Request Failed. \n'+ res;
+        }
+        token = result['access_token'];
+        token_type = result['token_type'];
+        if(!token || !token_type){
+            throw 'Request failed: missing token\n'+ res;
+        }
+    };
+
+    var fetchLoginEvents = function() {
+        var lastRun = getLastRun();
+        var date = new Date();
+        var now = {value : date.toISOString()};
+        if (!lastRun || !lastRun.value) {
+            date.setDate(date.getDate() - 1);
+            lastRun = {value: date.toISOString()};
+        }
+        request_string = 'https://graph.microsoft.com/beta/identityRiskEvents?filter=createdDateTime%20gt%20'+lastRun.value;
+        res = sendRequest(request_string, 'GET');
+        events = JSON.parse(res).value;
+
+        incidents = [];
+        events.forEach(function(event){
+            incidents.push({name: 'Risky Signin', rawJSON: JSON.stringify(event).replace(/\\"/g, '"')});
+        });
+        setLastRun(now);
+        return JSON.stringify(incidents);
+    };
+
+    var getRiskEvent = function(id){
+        request_string = 'https://graph.microsoft.com/beta/identityRiskEvents/'+id;
+        res = JSON.parse(sendRequest(request_string, 'GET'));
+        return {
+            Type: entryTypes.note,
+            HumanReadable: tableToMarkdown('Azure AD Risk Event', res),
+            HumanReadableFormat: formats.markdown,
+            Contents: res,
+            ContentsFormat: formats.json
+        };
+    };
+
+    getToken();
```

Comment:
using globals is confusing, why not returning the token here and than using it in the second command?